pkg/vm/stackitem/item.go

```diff
@@ -594,19 +606,35 @@ func (i ByteArray) TryInteger() (*big.Int, error) {
 
 // Equals implements the Item interface.
 func (i *ByteArray) Equals(s Item) bool {
-	if len(*i) > MaxByteArrayComparableSize {
+	var limit = MaxByteArrayComparableSize
+	return i.equalsLimited(s, &limit)
+}
+
+// equalsLimited compares ByteArray with provided stackitem using the limit.
+func (i *ByteArray) equalsLimited(s Item, limit *int) bool {
+	lCurr := len(*i)
+	if lCurr > *limit || *limit == 0 {
 		panic(errTooBigComparable)
 	}
-	if i == s {
-		return true
-	} else if s == nil {
-		return false
+	var comparedSize = 1
+	defer func() { *limit -= comparedSize }()
+
+	if s == nil {
+		return i == nil
```

Comment:
How can `i` be `nil` after `len(*i)`?

Reply:
I've added  the check that `i` is not `nil` in the start, just in case.